src/main/java/com/ra4king/circuitsim/gui/peers/gates/GatePeer.java

```diff
@@ -167,8 +168,33 @@ public final void paint(GraphicsContext graphics, CircuitState circuitState) {
 				
 				graphics.fillOval(x, y, GuiUtils.BLOCK_SIZE, GuiUtils.BLOCK_SIZE);
 				graphics.strokeOval(x, y, GuiUtils.BLOCK_SIZE, GuiUtils.BLOCK_SIZE);
+			} else if (hasNegatedInput) {
+				// Imitate how a wire is drawn in Wire.paint()
+				GuiUtils.setBitColor(graphics, circuitState, c.getLinkWires());
+				graphics.setLineWidth(2.0);
+
+				int dx = 0, dy = 0;
+				switch (getProperties().getValue(Properties.DIRECTION)) {
```

Comment:
Use enhanced switch:

```
switch (getProperties().getValue(Properties.DIRECTION)) {
	case WEST -> dx = -GuiUtils.BLOCK_SIZE;
	case EAST -> dx = GuiUtils.BLOCK_SIZE;
	case NORTH -> dy = -GuiUtils.BLOCK_SIZE;
	case SOUTH -> dy = GuiUtils.BLOCK_SIZE;
}
```

Reply:
Wow neat, I'm going to wake up one morning and Java will have become OCaml